### PR TITLE
docs(button): update sizing guidance

### PR DESCRIPTION
**Pull request by tay1orjones**
[Reported on slack](https://ibm-studios.slack.com/archives/C2K6RFJ1G/p1679511549115269)

#### Changelog

**Changed**

- Update sizing guidance to match sizing changes made in v11

#### Testing / Reviewing

- View the storybook mdx docs for Button in the deploy preview. The sizing section should make sense and reflect the current state of the system.

### CODE REVIEW

File: packages/react/src/components/Button/Button.mdx [hunk 1]
```diff
@@ -268,17 +268,16 @@ an element appear as a button control to a screen reader. Check out the
 ### Button `size`
 
 This prop specifies which size the `Button` should be rendered at. Valid values
-are `sm`, `lg`, `xl` and `default`. If no `size` is specified, it
-renders as `default`.
+are `sm`, `md`, `lg`, and `xl`. If no `size` is specified, it renders as `lg`.
```
Comment (tw15egan) [inline packages/react/src/components/Button/Button.mdx]:
```suggestion
are `sm`, `md`, `lg`, `xl`, and `2xl`. If no `size` is specified, it renders as `lg`.
```

Comment (tay1orjones) [reply packages/react/src/components/Button/Button.mdx]: Ah yes, thanks!

**Review by tw15egan — APPROVED** [review]
LGTM 👍 ✅

File: packages/react/src/components/Button/Button.mdx [hunk 2]
```diff
@@ -268,17 +268,16 @@ an element appear as a button control to a screen reader. Check out the
...
   Submit
-</Button> <Button size="lg">Submit</Button> <Button size="xl">Submit</Button>
+</Button> <Button size="xl">Submit</Button>
 
 ```jsx
 <Button>Submit</Button>
 <Button size="sm">Submit</Button>
-<Button size="lg">Submit</Button>
+<Button size="md">Submit</Button>
 <Button size="xl">Submit</Button>
```
Comment (fbarroso24) [inline packages/react/src/components/Button/Button.mdx:281]: @tay1orjones should we add a `2xl` example?

Comment (tay1orjones) [reply packages/react/src/components/Button/Button.mdx:281]: Added!

**Review by francinelucca — APPROVED** [review]

**Pull request merged**